flocker/acceptance/endtoend/test_installer.py

```diff
@@ -0,0 +1,319 @@
+        time_running = time.time() - start_time
+        if time_running > time_limit:
+            Message.new(
+                message='Timeout waiting for stack target status',
+                stack_id=stack_id,
+                target_status=target_status,
+                current_status=current_status,
+            ).write()
+            return False
+        else:
+            Message.new(
+                message='Waiting for stack target status',
+                stack_id=stack_id,
+                target_status=target_status,
+                current_status=current_status,
+                time_running=time_running,
+            ).write()
+
+            time.sleep(10)
+
+
+def create_cloudformation_stack():
+    """
+    """
+    # Request stack creation.
+    stack_name = CLOUDFORMATION_STACK_NAME + str(int(time.time()))
+    output = check_output(
+        ['aws', '--region', REGION, 'cloudformation', 'create-stack',
+         '--parameters', json.dumps(PARAMETERS),
+         '--stack-name', stack_name,
+         '--template-body', S3_CLOUDFORMATION_TEMPLATE]
+    )
+
+    output = json.loads(output)
+    stack_id = output['StackId']
+    Message.new(cloudformation_stack_id=stack_id)
+    stack_report = wait_for_stack_status(stack_id, 'CREATE_COMPLETE')
+    return stack_report
+
+
+def delete_cloudformation_stack(stack_id):
+    """
+    """
+    result = get_stack_report(stack_id)
+    outputs = result['Outputs']
+    s3_bucket_name = get_output(outputs, 'S3BucketName')
+    check_call(
+        ['aws', 's3', 'rb', 's3://{}'.format(s3_bucket_name), '--force']
+    )
+
+    check_call(
+        ['aws', '--region', REGION, 'cloudformation', 'delete-stack',
+         '--stack-name', stack_id]
+    )
+
+    return wait_for_stack_status(stack_id, 'DELETE_COMPLETE')
+
+
+def get_output(outputs, key):
+    for output in outputs:
+        if output['OutputKey'] == key:
+            return output['OutputValue']
+
+
+class DockerComposeTests(AsyncTestCase):
+    """
+    Tests for AWS CloudFormation installer.
+    """
+    run_tests_with = async_runner(timeout=timedelta(minutes=20))
+
+    def setUp(self):
+        stack_report = create_cloudformation_stack()
+        outputs = stack_report['Outputs']
+        self.stack_id = stack_report['StackId']
+        self.client_ip = get_output(outputs, 'ClientNodeIP')
+        self.agent_node_1 = get_output(outputs, 'AgentNode1IP')
+        self.agent_node_2 = get_output(outputs, 'AgentNode2IP')
+        self.control_node_ip = get_output(outputs, 'ControlNodeIP')
+        self.docker_host = self.control_node_ip + ':2376'
+        local_certs_path = self.mktemp()
+        check_call(
+            ['scp', '-o', 'StrictHostKeyChecking no', '-r',
+             'ubuntu@{}:/etc/flocker'.format(self.client_ip),
+             local_certs_path]
+        )
+        self.addCleanup(cleanup, self, local_certs_path)
+        d = maybeDeferred(super(DockerComposeTests, self).setUp)
+        return d
```

Comment:
Just returning the result of upcalling `setUp` is enough. No reason to wrap in `maybeDeferred`.